src/jit/bitsetasshortlong.h

```diff
@@ -664,9 +664,11 @@ void BitSetOps</*BitSetType*/ BitSetShortLongRep,
                /*BitSetTraits*/ BitSetTraits>::ClearDLong(Env env, BitSetShortLongRep& bs)
 {
     assert(!IsShort(env));
-    // Recall that ClearD does *not* require "bs" to be of the current epoch.
-    // Therefore, we must allocate a new representation.
-    bs = MakeEmptyArrayBits(env);
```

Comment:
This change makes me *very* nervous, as we do not seem to have any way to verify that the size of the bitset has not changed since we created it (i.e. there appears to be no way to check that the epoch at creation is the same as the epoch at the clear). I think you should introduce a new method (perhaps `ClearDUnsafe`? I don't have a great name for it) with the semantics you've introduced here.